qa/L0_metrics/histogram_metrics_test.py

```diff
@@ -0,0 +1,171 @@
+            if error:
+                user_data.append(error)
+            else:
+                user_data.append(result)
+
+        # list to hold the results of inference.
+        user_data = []
+
+        # Inference call
+        triton_client.start_stream(callback=partial(callback, user_data))
+        triton_client.async_stream_infer(
+            model_name=model_name,
+            inputs=inputs,
+            outputs=outputs,
+        )
+        triton_client.stop_stream()
+
+        # Wait until the results are available in user_data
+        time_out = 10
+        while (len(user_data) == 0) and time_out > 0:
+            time_out = time_out - 1
+            time.sleep(1)
+
+        # Display and validate the available results
+        if len(user_data) == 1:
+            # Check for the errors
+            if type(user_data[0]) == InferenceServerException:
+                print(user_data[0])
+                sys.exit(1)
+
+    def test_ensemble_decoupled(self):
+        ensemble_model_name = "ensemble"
+        wait_secs = 1
+
+        # Infer
+        inputs = []
+        outputs = []
+        inputs.append(grpcclient.InferInput("INPUT", [1], "FP32"))
+        outputs.append(grpcclient.InferRequestedOutput("OUTPUT"))
+
+        # Create the data for the input tensor. Initialize to all ones.
+        input_data = np.ones(shape=(1), dtype=np.float32) * wait_secs
+        # Initialize the data
+        inputs[0].set_data_from_numpy(input_data)
+
+        self.async_stream_infer(ensemble_model_name, inputs, outputs)
+
+        # Checks metrics output
+        first_response_family = "nv_inference_first_response_histogram_ms"
+        decoupled_model_name = "async_execute_decouple"
+        histogram_dict = self.get_histogram_metrics(first_response_family)
+
+        ensemble_model_count = get_histogram_metric_key(
+            first_response_family, ensemble_model_name, "1", "count"
+        )
+        ensemble_model_sum = get_histogram_metric_key(
+            first_response_family, ensemble_model_name, "1", "sum"
+        )
+        self.assertIn(ensemble_model_count, histogram_dict)
+        self.assertGreaterEqual(histogram_dict[ensemble_model_count], 1)
+        self.assertIn(ensemble_model_sum, histogram_dict)
+        self.assertGreaterEqual(
+            histogram_dict[ensemble_model_sum], 2 * wait_secs * MILLIS_PER_SEC
+        )
+
+        decoupled_model_count = get_histogram_metric_key(
+            first_response_family, decoupled_model_name, "1", "count"
+        )
+        decoupled_model_sum = get_histogram_metric_key(
+            first_response_family, decoupled_model_name, "1", "sum"
+        )
+        self.assertIn(decoupled_model_count, histogram_dict)
+        self.assertGreaterEqual(histogram_dict[decoupled_model_count], 1)
+        self.assertIn(decoupled_model_sum, histogram_dict)
+        self.assertGreaterEqual(
+            histogram_dict[decoupled_model_sum], wait_secs * MILLIS_PER_SEC
+        )
```

Comment:
Can you run the inference multiple times to show the `count` and `sum` are correctly aggregated, across multiple requests?

Reply:
Updated